src/main/java/frc/robot/subsystems/intake/commands/ToggleIntake.java

```diff
@@ -0,0 +1,21 @@
+package frc.robot.subsystems.intake.commands;
+
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import frc.robot.subsystems.intake.Intake;
+
+/**
+ * this command toggles the state of intake's pistons
+ * i.e. opened --> closed & closed --> opened
+ */
+public class ToggleIntake extends InstantCommand {
+    private Intake intake;
+    public ToggleIntake(Intake i) {

```

Comment:
Why do you shorten?

```suggestion
    public ToggleIntake(Intake intake) {
```
Also, maybe this command should be called `ToggleIntakeRetractor` to be precise.